keps/sig-architecture/2019-07-16-node-role-label-use.md

```diff
@@ -175,22 +147,54 @@ Some components like the external cloud provider controllers (considered to fall
 
 ## Design Details
 
+### Migrating existing deployments
+
+The proposed fixes will all require deployment-level changes. That must be staged across several releases, and it should be possible for deployers to move early and "fix" the issues that may be caused by their topology.
+
+Therefore, for each change we recommend the following process to adopt the new labels in successive releases:
+
+* Release 1 (1.16):
+  * Introduce a feature gate for disabling node-role being honored. The gate defaults to on. `LegacyNodeRoleBehavior=on`
+  * Define the new node label with an associated feature gate for each feature area. The gate defaults to off. `ServiceNodeExclusion=off` and `NodeDisruptionExclusion=off`
+  * Behavior for each functional area is defined as `(LegacyNodeRoleBehavior == on && node_has_role) || (FeatureGate == on && node_has_label)`
+  * No new components may leverage node-roles within Kubernetes projects.
+  * Early adopters may label their nodes to opt in to the features, even in the absence of the gate.
+  * The legacy alpha label `alpha.service-controller.kubernetes.io/exclude-balancer` is marked as deprecated
+* Release 2 (1.17):
+  * Deprecation of node role behavior in tree is announced for a future release as a warning to users
```

Comment:
recommend giving a specific target release in the announcement. "in a future release" makes it really hard to plan transitions.